### PR TITLE
[SparkR-209] `join`, `sort`, `filter` methods for DataFrame

### PR DESCRIPTION
**Pull request by cafreeman**
SQL-Like methods:
- `join`
- `sort`
- `orderBy`
- `filter`
- `where`

### CODE REVIEW

File: pkg/inst/tests/test_sparkSQL.R [hunk 1]
```diff
@@ -241,17 +241,75 @@ test_that("column calculation", {
...
+  filtered <- filter(df, "age > 20")
+  expect_true(count(filtered) == 1)
+  expect_true(collect(filtered)$name == "Andy")
+  filtered2 <- where(df, df$name != "Michael")
+  expect_true(count(filtered2) == 2)
+  expect_true(collect(filtered2)$age[2] == 19)
+})
+
+test_that("join() on a DataFrame", {
+df <- jsonFile(sqlCtx, jsonPath)
```
Comment (davies) [inline pkg/inst/tests/test_sparkSQL.R]: indent

**Comment by davies** [comment]
LGTM, just one comment about style.

**Comment by shivaram** [comment]
This is great. LGTM for me too. Merging this.

BTW can you update SPARKR-209 with what functions are still missing ?

**Comment by cafreeman** [comment]
Thanks :)

I'll update the JIRA now.

**Pull request merged**